examples/device/device.cpp

```diff
@@ -872,3 +903,227 @@ int CXX_QDMI_device_session_query_operation_property(
   }
   return QDMI_ERROR_NOTSUPPORTED;
 } /// [DOXYGEN FUNCTION END]
+
+int CXX_QDMI_device_session_query_telemetrysensor_property(
+    CXX_QDMI_Device_Session session, CXX_QDMI_TelemetrySensor telemetry_sensor,
+    QDMI_TelemetrySensor_Property prop, size_t size, void *value,
+    size_t *size_ret) {
+  if (session == nullptr || telemetry_sensor == nullptr ||
+      (value != nullptr && size == 0) ||
+      (prop >= QDMI_TELEMETRYSENSOR_PROPERTY_MAX &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM1 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM2 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM3 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM4 &&
+       prop != QDMI_TELEMETRYSENSOR_PROPERTY_CUSTOM5)) {
+    return QDMI_ERROR_INVALIDARGUMENT;
+  }
+  ADD_STRING_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_ID,
+                      telemetry_sensor->id.c_str(), prop, size, value, size_ret)
+  ADD_STRING_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_UNIT,
+                      telemetry_sensor->unit.c_str(), prop, size, value,
+                      size_ret)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_TELEMETRYSENSOR_PROPERTY_SAMPLINGRATE,
+                            long int, telemetry_sensor->sampling_rate.count(),
```

Comment:
```suggestion
                            int32_t, telemetry_sensor->sampling_rate.count(),
```
Just to stay consistent, I would prefer the equivalent type. `int32_t`.